applications/FluidDynamicsApplication/custom_processes/distance_modification_process.cpp

```diff
@@ -36,28 +36,51 @@ DistanceModificationProcess::DistanceModificationProcess(
     const bool CheckAtEachStep,
     const bool NegElemDeactivation,
     const bool RecoverOriginalDistance)
-    : Process(), mrModelPart(rModelPart) {
-
+    : Process(),
+      mrModelPart(rModelPart)
+{
+    // Member variables initialization
     mDistanceThreshold = DistanceThreshold;
     mCheckAtEachStep = CheckAtEachStep;
     mNegElemDeactivation = NegElemDeactivation;
     mRecoverOriginalDistance = RecoverOriginalDistance;
+
+    // Initialize the EMBEDDED_IS_ACTIVE variable flag to 0
+    this->InitializeEmbeddedIsActive();
 }
 
 DistanceModificationProcess::DistanceModificationProcess(
     ModelPart& rModelPart,
     Parameters& rParameters)
-    : Process(), mrModelPart(rModelPart)
+    : Process(),
+      mrModelPart(rModelPart)
 {
+    // Check default settings
     this->CheckDefaultsAndProcessSettings(rParameters);
+
+    // Initialize the EMBEDDED_IS_ACTIVE variable flag to 0
+    this->InitializeEmbeddedIsActive();
 }
 
 DistanceModificationProcess::DistanceModificationProcess(
     Model &rModel,
     Parameters &rParameters)
-    : Process(), mrModelPart(rModel.GetModelPart(rParameters["model_part_name"].GetString()))
+    : Process(),
+      mrModelPart(rModel.GetModelPart(rParameters["model_part_name"].GetString()))
 {
+    // Check default settings
     this->CheckDefaultsAndProcessSettings(rParameters);
+
+    // Initialize the EMBEDDED_IS_ACTIVE variable flag to 0
+    this->InitializeEmbeddedIsActive();
+}
+
+void DistanceModificationProcess::InitializeEmbeddedIsActive()
+{
+    for (int i_node = 0; i_node < static_cast<int>(mrModelPart.NumberOfNodes()); ++i_node) {
+        auto it_node = mrModelPart.NodesBegin() + i_node;
+        it_node->SetValue(EMBEDDED_IS_ACTIVE, 0);
```

Comment:
please use the variable utils here

Reply:
The `VariableUtils` has an OpenMP pragma so it would throw the same error that raised this issue.

Reply:
SetValue is threadsafe by design, while GetValue is not
=> #3712

Hence you can use it

Reply:
I tried and it thrown me an error in full debug mode. I'll check again.

Reply:
After reviewing the implementation, I guess that it is threadsafe once the variable has been allocated. However, it it is the first time you do the `SetValue`, meaning that the variable is to be allocated in such call, it is not threadsafe not because of the set operation but because of the variable allocation.

Maybe @roigcarlo and @RiccardoRossi can clarify this.

Reply:
Solved. @roigcarlo explained it to me.

Reply:
I had the exact same question, thats why I opened #3712

@RiccardoRossi explained there that BY DESIGN `SetValue` does not throw since one deliberately sets a value. `GetValue` however might SILENTLY allocate, which is sth that is not wanted in OMP

=> the datavaluecontainer itself is threadsafe, as long as it is not shared, which is not the case for the nodes, every node has it own

thats why the omp is safe in the variableutils

Reply:
Now the point is that there are no method to set an unsigned int variable to zero in the `Variable Utils` (or I can't find it).  Thus, I'd suggest to finish  #4723 first.